### PR TITLE
Adds tests for cli/set/sample

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR tests the cli set sample. N.B! the testing skips the lims-dependency when setting gender of the sample, please make sure this works on stage as described below!

How to test:
0. install cg on stage on rasta
1. us
1. run 'cg set sample [sample_id] -s [sex]
1. open stage-lims

Expected outcome:
The gender propagated to LIMS

also. Build successful
number of tests increased
coverage increased

### CODE REVIEW

File: cg/cli/set.py [hunk 1]
```diff
@@ -77,7 +76,8 @@ def sample(context, sex, customer, comment, downsampled_to, apptag, capture_kit,
         context.obj['status'].commit()
 
         print(click.style('update LIMS/Gender', fg='blue'))
-        lims_api.update_sample(sample_id, sex=sex)
+        if context.obj.get('lims'):
```
Comment (ingkebil) [inline cg/cli/set.py]: Could you explain why this is behind an if statement?

File: tests/cli/test_cli_set_sample.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+    sample_id = 'dummy_sample_id'
+    result = invoke_cli(['--database', disk_store.uri, 'set', 'sample', sample_id])
+
+    # THEN then it should complain on invalid sample
+    assert result.exit_code == 1
+
+
+def test_set_sample_sex(invoke_cli, disk_store: Store):
+    # GIVEN a database with a male sample
```
Comment (ingkebil) [inline tests/cli/test_cli_set_sample.py]:
```suggestion
    # GIVEN a database with a female sample
```

File: tests/cli/test_cli_set_sample.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    # GIVEN a database with a sample and two applications
+    sample_id = add_sample(disk_store).internal_id
+    application_tag = ensure_application_version(disk_store, 'another_application').application.tag
+    assert disk_store.Sample.query.first().application_version.application.tag != application_tag
+
+    # WHEN calling set sample with an invalid application
+    result = invoke_cli(['--database', disk_store.uri, 'set', 'sample', sample_id, '-a',
+                         application_tag])
+
+    # THEN then it should complain about missing application instead of setting the value
```
Comment (ingkebil) [inline tests/cli/test_cli_set_sample.py]:
```suggestion
    # THEN then the application should have been set
```

File: cg/cli/set.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/cli/set.py]: I don't know how test the LIMS part so if it is not initialized with LIMS we skip trying to send the gender to LIMS. Do you have any suggestion how we could do it?

**Pull request merged**